core/src/com/unciv/ui/utils/ExtensionFunctions.kt

```diff
@@ -360,15 +360,25 @@ fun (() -> Unit).wrapCrashHandlingUnit(
     return { wrappedReturning() ?: Unit }
 }
 
-/** For filters containing '{', apply the [predicate] to each part inside "{}" and aggregate using [operation];
- *  otherwise return `null` for Elvis chaining of the individual filter. */
-fun <T> String.filterCompositeLogic(predicate: (String) -> T?, operation: (T, T) -> T): T? {
-    val elements: List<T> = removePrefix("{").removeSuffix("}").split("} {")
+private val compositeLogicCheckRegex = Regex("""^\s*(?:\{[^{}]+}\s*)+$""")
+private val compositeLogicSplitRegex = Regex("""\{([^{}]+)}""")
+/** For filters containing '{', apply the [predicate] to each part inside "{}" and aggregate using [operation].
+ * 
+ *  A syntax check is performed, if the string doesn't consist only of non-empty, non-nested {} pairs separated by optional whitespace, then [invalidSyntaxResult] is returned. 
+ *  If valid, this returns `null` if and only if all [predicate] results were `null`, otherwise the aggregation by the [operation] lambda of non-null [predicate] results. */
+fun <T> String.filterCompositeLogic(
+    predicate: (String) -> T?,
+    invalidSyntaxResult: T? = null,
```

Comment:
Same for this, if functions that are always called with the same value for a parameter that probably means the parameter is unneeded, YAGNI etc.

Reply:
Waitasec
Are you planning to use this function somewhere else? Because maybe this should just be collapsed into filterCompositeCheckErrors

Reply:
Two values: `RulesetInvariant` and `false`

... and yes the vision that it could be useful for future expansions is fuzzy but existing. For example, I still imagine filters with 'or' capability would in some cases improve in-game readability quite a bit - but for that the templating function would need to use parameterized RegExes as well. It pains me two regexes are needed, but alas this is not mono, or I haven't found the way.